tests/integration/cmd_push_test.go

```diff
@@ -0,0 +1,173 @@
+package integration
+
+import (
+	"github.com/openshift/odo/tests/helper"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("odoCmdPush", func() {
+	var project string
+	var context string
+
+	appName := "app"
+	cmpName := "nodejs"
+
+	oc = helper.NewOcRunner("oc")
+
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		project = helper.CreateRandProject()
+		context = helper.CreateNewContext()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.DeleteProject(project)
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("when push command is executed", func() {
+		It("should not build when no changes are detected in the directory and build when a file change is detected", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context+"/nodejs-ex", "--app", appName)
+
+			helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+
+			helper.CmdShouldPass("odo", "url", "create", "--port", "8080", "--context", context+"/nodejs-ex")
+
+			helper.CmdShouldPass("ls", context+"/nodejs-ex/.odo", "-a")
+
+			helper.CmdShouldPass("cat", context+"/nodejs-ex/.odo/odo-file-index.json")
```

Comment:
Won't work on windows environment. Can you please use go library instead of ```cat``` , ```ls```

Reply:
these are for debugging, will remove after they pass